app/services/notifications/mail_service.rb

```diff
@@ -26,45 +26,51 @@
 # See COPYRIGHT and LICENSE files for more details.
 #++
 
-class Notifications::MailService
-  def initialize(notification)
-    self.notification = notification
-  end
+module Notifications
+  class MailService
+    include WithMarkedNotifications
 
-  def call
-    return unless supported?
-    return if ian_read?
+    def initialize(notification)
+      self.notification = notification
+    end
 
-    strategy.send_mail(notification)
-  end
+    def call
+      return unless supported?
+      return if ian_read?
 
-  private
+      with_marked_notifications(notification.id) do
+        strategy.send_mail(notification)
+      end
```

Comment:
This works within this PR but will break the reminder mail functionality for "mentioned" notifications once #9700 is merged.

For mentioned notifications, if the user chooses to do so, three messages can be sent to the user:
* An in app notification (not relevant for the discussion)
* An individual email "You are mentioned"
* A reminder in which the mentioned notification is included

In the later, only notifications that have `sent_mail` set to `false` are included. Before this PR, there was a distinction between `read_mail` and `read_mail_digest`. If we now map both of those to `sent_mail`, the reminder mail will not contain "mentioned" notifications if the user opted to have them sent directly. 

I only have two not so great ideas:
* continue to have two columns, one for the reminder and one for individual mail 
* don't maintain the `sent_mail` status for individual mails since we do not depend on it and only use it for reminder mails

Reply:
I'll add two columns as I prefer to be sure what happened rather than infer from the state machine. In case the workflow job ever fails before sending, we could find out those notifications by filtering for one of the two mail columns.

Reply:
:+1: